src/views/CountryRiskWatch/HistoricalDataChart/index.tsx

```diff
@@ -0,0 +1,446 @@
+    const { countryId } = props;
+    const strings = useTranslation(i18n);
+    const [disasterFilter, setDisasterFilter] = useInputState<number | undefined>(undefined);
+    const chartContainerRef = useRef<HTMLDivElement>(null);
+    const chartBounds = useSizeTracking(chartContainerRef);
+
+    const query = useMemo<GoHistoricalQuery>(
+        () => ({ countries: [countryId] }),
+        [countryId],
+    );
+    const { response: historicalDataResponse } = useRequest<GoHistoricalResponse>({
+        skip: isNotDefined(countryId),
+        url: '/api/v2/go-historical/',
+        query,
+    });
+
+    const disasterOptions = unique(
+        historicalDataResponse?.results?.map(
+            (event) => {
+                if (!isValidDisaster(event.dtype)) {
+                    return undefined;
+                }
+
+                return {
+                    id: event.dtype.id,
+                    name: event.dtype.name,
+                };
+            },
+        ).filter(isDefined) ?? [],
+        (option) => option.id,
+    );
+
+    const filteredEvents = historicalDataResponse?.results?.filter(
+        (event) => {
+            if (!isValidDisaster(event.dtype)) {
+                return false;
+            }
+
+            if (isDefined(disasterFilter) && disasterFilter !== event.dtype.id) {
+                return false;
+            }
+
+            return true;
+        },
+    );
+
+    const maxValue = useMemo(
+        () => {
+            if (!filteredEvents) {
+                return 0;
+            }
+
+            const numAffectedList = filteredEvents.map(getNumAffected);
+            return Math.max(...numAffectedList);
+        },
+        [filteredEvents],
+    );
+
+    const yScale = useMemo(
+        () => {
+            const yBounds = getPrettyBreakableBounds(
+                { min: 0, max: maxValue },
+            );
+
+            return getScaleFunction(
+                yBounds,
+                { min: 0, max: chartBounds.height },
+                { start: chartMargin.top, end: chartMargin.bottom },
+                true,
+            );
+        },
+        [maxValue, chartBounds],
+    );
+
+    const xScale = useMemo(
+        () => (
+            getScaleFunction(
+                { min: 0, max: 12 },
+                { min: 0, max: chartBounds.width },
+                { start: chartMargin.left, end: chartMargin.right },
+            )
+        ),
+        [chartBounds],
+    );
+
+    const chartData = useMemo(
+        () => {
+            if (!filteredEvents) {
+                return undefined;
+            }
+
+            return filteredEvents.map(
+                (event) => {
+                    const disasterDate = new Date(event.disaster_start_date);
+                    const date = disasterDate.getDate();
+                    const month = disasterDate.getMonth();
+                    disasterDate.setMonth(month + 1);
+                    disasterDate.setDate(0);
```

Comment:
Are we moving the date to the last day of the month? Let's add a NOTE or even a helper function.

Reply:
Yes